### PR TITLE
Implement first user story functionality

### PR DESCRIPTION
**Pull request by Flott2410**
- form with selecting two countries on landing page
- route/controller/views for it
- show page for trip and for country
- needed to change trips that user is also allowed to not exist
**SO db:drop db:create db:migrate db:seed for everyone after merging**

### CODE REVIEW

File: app/controllers/trips_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    # assign user_id to trips if user is logged in
+    @trip.user = current_user
+    if @trip.save
+      redirect_to trip_path(@trip)
+    else
+      render 'pages/home'
+    end
+  end
+
+  def show
```
Comment (kleyv) [inline app/controllers/trips_controller.rb:16]: Don't create unnecessary methods. Keep it simple.

File: app/models/trip.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Trip < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, optional: true
```
Comment (kleyv) [inline app/models/trip.rb:2]: Why is this optional? Why would we have trips that are not assigned to users?

File: app/controllers/trips_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+before_action :set_trip, only: [ :show ]
+
+  def create
+    @trip = Trip.new(trip_params)
+    # assign user_id to trips if user is logged in
+    @trip.user = current_user
+    if @trip.save
+      redirect_to trip_path(@trip)
+    else
+      render 'pages/home'
```
Comment (kleyv) [inline app/controllers/trips_controller.rb:12]: Shouldn't it redirect to the path for trips#show?

File: config/routes.rb [hunk 4]
```diff
@@ -2,4 +2,6 @@
   devise_for :users
   root to: 'pages#home'
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :trips, only: [ :create, :show, :index ]
```
Comment (kleyv) [inline config/routes.rb:5]: I don't see how :index is necessary for the purpose of the story of this pull request.

File: db/migrate/20200831144533_create_trips.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@ def change
     create_table :trips do |t|
       t.references :origin, null: false, foreign_key: { to_table: :countries }
       t.references :destination, null: false, foreign_key: { to_table: :countries }
-      t.references :user, null: false, foreign_key: true
+      t.references :user, null: true, foreign_key: true
```
Comment (kleyv) [inline db/migrate/20200831144533_create_trips.rb:6]: Why are we allowing trips to not have users?

File: db/schema.rb [hunk 6]
```diff
@@ -52,7 +52,7 @@
   create_table "trips", force: :cascade do |t|
     t.bigint "origin_id", null: false
     t.bigint "destination_id", null: false
-    t.bigint "user_id", null: false
+    t.bigint "user_id"
```
Comment (kleyv) [inline db/schema.rb:55]: Why are we allowing trips to not have users?

**Review by kleyv — COMMENTED** [review]
Great work @Flott2410 @Tiago-Palhoca 
Regarding making users optional, is it so that visitors can also create the trips?

File: app/controllers/trips_controller.rb (hunk 1 above)
Comment (Flott2410) [reply app/controllers/trips_controller.rb:16]: Hey Kleyv. Exactly. Everyone should be able to create a trip and see the country page. But when bookmark or notification a visitor has to login first to become a user

File: app/models/trip.rb (hunk 2 above)
Comment (Flott2410) [reply app/models/trip.rb:2]: exactly. otherwise it would fail

File: db/schema.rb (hunk 6 above)
Comment (Tiago-Palhoca) [reply db/schema.rb:55]: We only need a registred user to bookmark or add notifications.
Every other action is open

File: app/controllers/trips_controller.rb (hunk 3 above)
Comment (Flott2410) [reply app/controllers/trips_controller.rb:12]: exactly. this is the path for trip show. check routes!!!!

File: config/routes.rb (hunk 4 above)
Comment (Flott2410) [reply config/routes.rb:5]: no it`s not. we forgot to delete. it. Was just for testign purposes

File: db/migrate/20200831144533_create_trips.rb (hunk 5 above)
Comment (Flott2410) [reply db/migrate/20200831144533_create_trips.rb:6]: because you also want the app to work for visitors. See above

File: db/schema.rb (hunk 6 above)
Comment (Flott2410) [reply db/schema.rb:55]: see above

**Pull request merged**